From bbe199b27aa52fc407db3372af00c3e9ffa20d1d Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Wed, 16 Feb 2022 13:51:42 -0300
Subject: [PATCH] microblaze: Use the correct select syscall (BZ #28883)

On Microblaze only __NR_newselect is implemented, even though kernel
advertise __NR_select on asm/unistd.h.  Since microblaze is the
only architecture that undef __ASSUME_PSELECT, the generic code
change is simpler than chaging the architecture syscall number.

Upstream-Status: Backport [https://sourceware.org/git/?p=glibc.git;a=commit;h=bbe199b27aa52fc407db3372af00c3e9ffa20d1d]

Acked-by: Mark Hatle <mark.hatle@xilinx.com>
Signed-off-by: Mark Hatle <mark.hatle@xilinx.com>
---
 sysdeps/unix/sysv/linux/select.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysdeps/unix/sysv/linux/select.c b/sysdeps/unix/sysv/linux/select.c
index 3bc8b0cc3d..a3f0a2eba7 100644
--- a/sysdeps/unix/sysv/linux/select.c
+++ b/sysdeps/unix/sysv/linux/select.c
@@ -108,7 +108,7 @@ __select64 (int nfds, fd_set *readfds, fd_set *writefds, fd_set *exceptfds,
       ptv32 = &tv32;
     }
 
-  int r = SYSCALL_CANCEL (select, nfds, readfds, writefds, exceptfds, ptv32);
+  int r = SYSCALL_CANCEL (_newselect, nfds, readfds, writefds, exceptfds, ptv32);
   if (timeout != NULL)
     *timeout = valid_timeval_to_timeval64 (tv32);
   return r;
-- 
2.17.1

